### ISSUE TITLE
modbus to USB adaptor installation

### ISSUE THREAD

**Issue by paultanner** [openenergymonitor/emonhub#172]
I have followed the config guidance and tried this
```
[[MBUS]]
    Type = EmonHubMBUSInterfacer
    [[[init_settings]]]
        device = /dev/ttyAMA0
        baud = 2400
    [[[runtimesettings]]]
        pubchannels = ToEmonCMS,
        read_interval = 10
        validate_checksum = False
        nodename = MBUS
        [[[[meters]]]]
            [[[[[sontex]]]]]
                address = 1
                type = sontex531
```
The device (from emon shop) is plugged into a USB port on a pi and the A port of this is connected to the MBUS port of my Sontex using a piece of telephone wire.

The error I get is
```
MainThread Device communication error - check settings                         
```
There may be some other config to do, maybe of the Sontex?
Any suggestions please.

**Comment by glynhudson** [comment]
Hi Paul, have you checked to see see what mbus address the sontex is set to? Trystan has made a script to check the address and change the address if needed: https://github.com/emoncms/usefulscripts/tree/master/mbus

**Comment by paultanner** [comment]
@glyn sent me this https://github.com/emoncms/usefulscripts/tree/master/mbus and I also saw that I was using a serial port not /dev/ttyUSB0.
I modified the check script to try all 254 possible addresses but the meter did not respond on any of them.
Any further suggestions?

**Comment by glyn** [comment]
> @glyn sent me this https://github.com/emoncms/usefulscripts/tree/master/mbus and I also saw that I was using a serial port not /dev/ttyUSB0. I modified the check script to try all 254 possible addresses but the meter did not respond on any of them. Any further suggestions?

Did you mean @glynhudson ?

**Comment by paultanner** [comment]
Yes @glynhudson.  Forgot to check before submitting.
I have since found that the sontex is set to 2400 baud and address 0.
Am trying to find out how to change that as the sontex doc is unclear.

**Comment by glynhudson** [comment]
You can change the Sontex Mbus address using this script: https://github.com/emoncms/usefulscripts/blob/master/mbus/mbus_set_address.py

**Comment by paultanner** [comment]
The problem is that the meter does not respond at all.  The device address is set to zero.  I have asked Sontex if I can change that setting another way but have yet to hear back from them.

**Comment by paultanner** [comment]
I have now set the meter address to 1. Still no response.

**Comment by glynhudson** [comment]
It sounds like you're doing everything correct! Mbus on Sontex has just worked for me assuming the address was set correctly. I'll ask @TrystanLea to take a look, he's the mbus expert, and he wrote the above scripts to set the address and read from the meter.

**Comment by paultanner** [comment]
I'm wondering what other settings might be needed?  baud rate looks OK (2400). pi is connected on /dev/ttyUSB0.

**Comment by paultanner** [comment]
Yes. Stockshed (Sontex dist) confirm that power is required.  Now I need to know how to do that.  I can run AC power to it but do not have the necessary info to hook it up.  I gather that m-bus needs +/- 15v to operate. Sounds like I should have got the mains-powered version as that might have this other supply rail?

**Comment by paultanner** [comment]
Correction: apparently the normal thing is for the data logger to power the m-bus, not the meter (as there may be multiple data sources on the m-bus).  Looks like I need to adapt the m-bus to usb to add the power circuit.  Any suggestions on that?  I'm happy to do some breadboarding/ soldering.

**Comment by TrystanLea** [comment]
Hello Paul. The MBUS reader already has an integrated power supply for the MBUS part, but it doesn't power the heat meter. That requires a separate AC power supply for the Sontex itself.

**Comment by paultanner** [comment]
Thx Trystan. When I got the Sontex there was a choice of battery or mains power.  I chose battery and the meter is working.  Are you saying that the mains option powers the m-bus whereas the battery does not?

**Comment by paultanner** [comment]
Also, when I look at your m-bus to usb adaptor I don't see the kind of components that would supply the voltage for the m-bus.  So I'm wondering how you test it.  If this involves a heat meter then which model exactly.

**Comment by TrystanLea** [comment]
Hello Paul, do you have this part?

![image](https://user-images.githubusercontent.com/503186/169605428-8b5df7f8-fb84-4b17-9015-8456185c492a.png)

The small inductor L1, chip with 5 pins, diode and other components form a boost converter producing +34V. You should be able to measure 32-34V across the capacitor nearest the AA terminals (between A and Ground). Let me know what voltage you see and what voltage you see across A-B as well?

I test them in the office against a Qalcosonic E3, just a heat meter that I have free here.

**Comment by paultanner** [comment]
Yes, that's the board I have. I get 32.8v between A and GND.  31.3v between A and B.  The cable is only about 2 m. Seems like lack of power is not the problem.

In mbus_check_address.py there are no checks other than sending the prompt and receiving nothing back.  I'm wondering what else I could do to find the problem?

**Comment by TrystanLea** [comment]
Hello Paul, Im a little confused on the power supply issue/requirement for the Sontex531. I remember that being a requirement in the past but there's nothing in the installation manual that suggests it should not work on battery power. It might be worth us sending you another MBUS reader to try, it is quite easy to damage the reader if the A-B terminals are shorted.

**Comment by paultanner** [comment]
Thx Trystan,
I've been wondering if I could devise something akin to a loopback test.
Maybe just send frames continuously to make sure the sender part is working.  Maybe then use a switch and resistor test the receiver part.  Could be a good idea before shipping another board, especially as we can't be sure that the meter's m-bus feature is working.  Another idea would be to get hold of a cheap m-bus slave.

**Comment by glynhudson** [comment]
Hi Paul, any progresses? We're more than happy to send a replacement reader to rule out any issues with the reader. Please send an email to support@openenergymonitor.zendesk.com to arrange this.
